### ISSUE TITLE
UWP could not load library

### ISSUE THREAD

**Issue by JKennedy24** [adamfisher/Xamarin.Forms.VideoPlayer#99]
UWP project using FallCreators update as min and target version
throws error when initialising a page:

> Could not load file or assembly 'Octane.Xamarin.Forms.VideoPlayer, Version=2.1.0.0, Culture=neutral, PublicKeyToken=null'. The located assembly's manifest definition does

**Comment by JKennedy24** [comment]
Probably caused because the PCL version is still on `V2.0.0.0`. should be `V2.1.0.0`

**Comment by adamfisher** [comment]
This will be addressed in the next release. 

As a workaround for now you should be able to double-click the warning in Visual Studio and have it auto-generate an assembly binding in your `App.config` that looks something like the following.

Alternatively, you can also create one manually if you don't have an app.config and will need to add it to the main project folder (Build Action: None, Copy to Output Directory: Do not copy).

```
<?xml version="1.0" encoding="utf-8" ?>
<configuration>
  <runtime>
    <assemblyBinding xmlns="urn:schemas-microsoft-com:asm.v1">
      <dependentAssembly>
        <assemblyIdentity name="Octane.Xamarin.Forms.VideoPlayer.UWP" culture="neutral" />
        <bindingRedirect oldVersion="0.0.0.0-2.1.0.0" newVersion="2.1.0.0" />
      </dependentAssembly>
    </assemblyBinding>
  </runtime>
</configuration>
```

**Comment by JKennedy24** [comment]
@adamfisher While using my automated build for UWP I was also getting errors like this:

> 2018-02-05T10:28:51.5520747Z   C:\Program Files (x86)\Microsoft SDKs\NuGetPackagesFallback\microsoft.net.native.compiler\1.7.3\tools\Microsoft.NetNative.targets(697,5): error : Type 'System.IO.Stream' was not included in compilation, but was referenced in method 'StreamVideoSourceHandler.GetFileName(Stream, string)'. There may have been a missing assembly. 
> 2018-02-05T10:28:51.5521148Z   C:\Program Files (x86)\Microsoft SDKs\NuGetPackagesFallback\microsoft.net.native.compiler\1.7.3\tools\Microsoft.NetNative.targets(697,5): error : Type 'System.Threading.Tasks.Task' was not included in compilation, but was referenced in method 'VideoPlayerRenderer.UpdateSource(VideoPlayer)'. There may have been a missing assembly. 
> 2018-02-05T10:28:51.5521580Z   C:\Program Files (x86)\Microsoft SDKs\NuGetPackagesFallback\microsoft.net.native.compiler\1.7.3\tools\Microsoft.NetNative.targets(697,5): error : Type 'System.Runtime.CompilerServices.ConfiguredTaskAwaitable`1' was not included in compilation, but was referenced in type 'Octane.Xamarin.Forms.VideoPlayer.UWP.SourceHandlers.StreamVideoSourceHandler+<LoadVideoAsync>d__0'. There may have been a missing assembly.

Do these errors look linked or is this something else?

Does Octane Video player require a minimum of `.Net Standard 2.0`?

**Comment by ozzy1873** [comment]
Adding the app.config file does not seem to help:

Could not load file or assembly 'Octane.Xamarin.Forms.VideoPlayer, Version=2.1.0.0, Culture=neutral, PublicKeyToken=null'. The located assembly's manifest definition does not match the assembly reference. (Exception from HRESULT: 0x80131040)

The app.config files for other project have a PublicKeyToken. Should that be included?

**Comment by hauteur** [comment]
This also didn't work for me.  I did notice that my error is somewhat different from the OP.  Mine says:

System.IO.FileNotFoundException: 'Could not load file or assembly 'Octane.Xamarin.Forms.VideoPlayer, Version=2.1.0.0, Culture=neutral, PublicKeyToken=null'. **The system cannot find the file specified.'**

When are you expecting to drop the new release?

**Comment by hauteur** [comment]
I downloaded the sample, and it gets the same error upon loading.

Also - not trying to be rude, but is anyone monitoring this?  I have really got to get this working very very soon or I need to move on to a different library.

**Comment by JKennedy24** [comment]
This component straight up hasn't worked on UWP for 4 months. not great for a paid component

@hauteur an alternative is: https://www.nuget.org/packages/Rox.Xamarin.Video/

and you don't have to pay for it

**Comment by hauteur** [comment]
Thank you for that @JKennedy24.  Better to know how well it's supported before I go into production, I guess.

**Comment by adamfisher** [comment]
The UWP project currently targets the Fall Creators Update.

![image](https://user-images.githubusercontent.com/1834711/40588363-6079f0ee-61aa-11e8-8d14-552c6e2b52aa.png)

I think I see the problem in the NuGet package. The netstandard1.4 targeted version of the `Octane.Xamarin.Forms.VideoPlayer` shared library is correctly specifying 2.1.0 in the DLL properties whereas the one inside the UAP targeted version contains 0.0.0 for that same DLL.

**Comment by adamfisher** [comment]
Improvements and/or fixes have been made in version 2.2.0. Please see the [release notes](https://github.com/adamfisher/Xamarin.Forms.VideoPlayer/releases/tag/2.2.0) for more information. The NuGet package should be published shortly.

**Comment by adamfisher** [comment]
Some additional testing with UWP and .NET Standard shows the new .NET Standard platform introduced a new way to call `Assembly.GetExecutingAssembly()`, `Assembly.GetEntryAssembly()`, and `Assembly.GetCallingAssembly()` which is used to locate video sources. A fix has been identified for this issue as well. Basically if you try to set an embedded resource as the source of a video, you would get an exception because .NET standard does things a little differently. Unfortunately this will mean pushing .NET standard support up from 1.4 to 2.0.

The related details to the changes in Xamarin.Forms can be found from this PR for anyone interested: https://github.com/xamarin/Xamarin.Forms/pull/1369

**Comment by adamfisher** [comment]
Improvements and/or fixes have been made in version 2.2.0. Please see the [release notes](https://github.com/adamfisher/Xamarin.Forms.VideoPlayer/releases/tag/2.2.0) for more information. The NuGet package should be published shortly.
